### ISSUE TITLE
When navigating away and back to a page asControl instantly fails if page loads too slow

### ISSUE THREAD

**Issue by philippthiele** [ui5-community/wdi5#512]
**Describe the bug**

This is the step implementation, where the first "asControl" fails:
```
When(/^In the dropdown for filters on <(.*)> I select value <(.*)>/, async function (dropdown, itemValue) {
    console.log(`In the dropdown for filters on ${dropdown} I select value ${itemValue}`);
    await browser.asControl({
        forceSelect: true,
        selector: {
            id: dropdown,
            viewName: "com.sap.calm.uam.um.view.UserDetails"
        }
    }).open();
    await browser.asControl({
        forceSelect: true,
        selector: {
            controlType: "sap.m.StandardListItem",
            viewName: "com.sap.calm.uam.um.view.UserDetails",
            properties: {
                title: itemValue
            },
            searchOpenDialogs: true
        }
    }).press();
});

```

This step implementation works well more than once before the actual failure and sometimes also in the problematic situation it works, but I think when the navigation is too slow, this issue might be happening.

Here are the logs:
![image](https://github.com/ui5-community/wdi5/assets/1737722/4d6fdea1-1aef-470e-b867-942028d6a515)

In this situation this is what happens:
1. We are on page `/index.html?sap-language=EN#user-display&/userList/OneColumn`
2. We navigate away to page `/index.html?sap-language=EN#user-display` <- here the log I posted above starts
3. We navigate back to `/index.html?sap-language=EN#user-display&/userList/OneColumn`
4. The first asControl, which should return a dropdown, with the `open()` command is executed, but fails already after the first try without waiting
5. The access to `StandardListItem` fails which confirms that it only tried very shortly to get the dropdown and immediately moved forward to the next step

**Expected behavior**
wdi5 tries to retrieve the control **until the timeout runs out** and if it does not work, it should end the execution and not execute the next step

**Runtime Env (please complete the following information):**

- `wdi5/wdio-ui5-service`-version: "1.5.1
- `UI5` version: 1.116.0
- `wdio`-version (output of `wdio --version`): 7.31.1
- `node`-version (output of `node --version`): v18.14.1
- OS: [e.g. Linux openSuSE]
- Browser + Version chrome 111.0.0

**Additional context**
Add any other context about the problem here,
e.g. any options the target browser is started with like `--headless` or
if the tests run in a CI environment

```
        'goog:chromeOptions': {
            args: [
                "--window-size=1920,1080",
                "--lang=en-US",
                "--headless",
                "--disable-gpu",
                "--disable-extensions",
                "--disable-translate",
                "--disable-dev-shm-usage",
                "--no-sandbox",
                "--disable-popup-blocking"
            ]
```

**Comment by Siolto** [comment]
Hi @philippthiele,

wdi5 has no polling mechanism like OPA5 has. We do not try to retrieve the control till the timeout reaches. We just try to identify it **once**. As wdi5 is synchronized with the UI5 Framework we wait with the help of the autowaiter till all synchronous work is done and only then try to identify the control. Maybe we have a problem with the autowaiter again.

Could we have a look at your issue together?

**Comment by philippthiele** [comment]
@Siolto ok that explains why it does not wait for the element, unfortunately this only happens during ci-cd when multiple tests run in parallel, why I assume it loads probably a little slower than locally. I just tried to add screenshots before and after each step, and now the step is green, I suppose that the screenshots give the page more time to load and thus avoid this error. I am at a loss how we can reproduce the error and figure out how to fix it 😣

**Comment by Siolto** [comment]
In [Volkers session](https://youtu.be/aCyMC27ED80?t=12693) at the UI5Con (you also got a [shoutout during his talk](https://youtu.be/aCyMC27ED80?t=13423) 😄) he showed how you can execute your tests with [throttled network](https://youtu.be/aCyMC27ED80?t=12882). Maybe you could reproduce it with throttled network?

**Comment by philippthiele** [comment]
@Siolto is it possible this only works with the devtools protocol? 
https://v7.webdriver.io/docs/api/browser/throttle/
When I try to run our tests with the devtools protocol by setting `automationProtocol: 'devtools'` in our wdio.conf.js, I get an error
```
[0-0] 2023-08-22T07:31:28.114Z ERROR @wdio/runner: FetchError: Failed to fetch browser webSocket URL from http://localhost:51877/json/version: request to http://localhost:51877/json/version failed, reason: connect ECONNREFUSED ::1:51877
[0-0]     at ClientRequest.<anonymous> (C:\SAPDevelop\GitHub\uam-userservice-ui\node_modules\node-fetch\lib\index.js:1491:11)
[0-0]     at ClientRequest.emit (node:events:513:28)
[0-0]     at ClientRequest.emit (node:domain:489:12)
[0-0]     at Socket.socketErrorListener (node:_http_client:502:9)
[0-0]     at Socket.emit (node:events:513:28)
[0-0]     at Socket.emit (node:domain:489:12)
[0-0]     at emitErrorNT (node:internal/streams/destroy:151:8)
[0-0]     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
[0-0]     at processTicksAndRejections (node:internal/process/task_queues:82:21)
```

Upgrading wdio to v8 or downgrading node might help, if I read this issue correctly: https://github.com/webdriverio/webdriverio/issues/8279

But we are currently running with node 18 and we are set to wdio v7 since wdi5 right now only supports wdio v7. I would postpone solving this issue until I can reproduce the issue somehow locally or someone else happens to run into it and can reproduce it.

**Comment by vobu** [comment]
on a sidenote, `browser.throttle` seems to work fine in `wdi5^2` and `wdio^8` without specifically configuring the `devtools` protocol.

From my UI5con session:
```js
// excerpt from wdio.conf.js
export const config = {
// ...

  services: ["chromedriver", "ui5"],

  specs: ["./test/**/*.test.js"],
  maxInstances: 1,
  capabilities: [
    {
      maxInstances: 1,
      browserName: "chrome",
      acceptInsecureCerts: true
    }
  ]
//...
}

// Basic.test.js
import { wdi5 } from "wdio-ui5-service" // this is 2.0-alpha
// ... 
describe("wdi5 can do that?!", async () => {
  it.only("should throttle the network throughput for testing the 'busy' indicator", async () => {
    await browser.throttle("Good2G") // <--
    await wdi5.goTo("#/Other")
    const list = await browser.asControl(selectors.peopleList)
    const people = await list.getItems(true)
    expect(people.length).toBe(9)
  })
// ...
```

```shell
 "spec" Reporter:
------------------------------------------------------------------
[chrome 115.0.5790.114 mac os x #0-0] Running: chrome (v115.0.5790.114) on mac os x
[chrome 115.0.5790.114 mac os x #0-0] Session ID: 5519073f6a62554605337e6613d4614a
[chrome 115.0.5790.114 mac os x #0-0]
[chrome 115.0.5790.114 mac os x #0-0] » /test/Basic.test.js
[chrome 115.0.5790.114 mac os x #0-0] wdi5 can do that?!
[chrome 115.0.5790.114 mac os x #0-0]    ✓ should throttle the network throughput for testing the 'busy' indicator
[chrome 115.0.5790.114 mac os x #0-0]
[chrome 115.0.5790.114 mac os x #0-0] 1 passing (11.9s)
```

I know this doesn't help with `wdi5^1` probably, but just adding context (and hopes for `wdi5^2 `😸) to this thread.

**Comment by vobu** [comment]
still relevant, will come back to this
